src/Testcontainers.Kusto/KustoBuilder.cs

```diff
@@ -0,0 +1,100 @@
+namespace Testcontainers.Kusto;
+
+using System.Threading.Tasks;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+/// <remarks>
+/// Builds a container running the "Azure Data Explorer Kusto emulator":
+/// https://learn.microsoft.com/en-us/azure/data-explorer/kusto-emulator-overview
+/// </remarks>
+[PublicAPI]
+public sealed class KustoBuilder : ContainerBuilder<KustoBuilder, KustoContainer, KustoConfiguration>
+{
+    public const string KustoImage = "mcr.microsoft.com/azuredataexplorer/kustainer-linux:latest";
+
+    public const ushort KustoPort = 8080;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KustoBuilder" /> class.
+    /// </summary>
+    public KustoBuilder()
+        : this(new KustoConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KustoBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private KustoBuilder(KustoConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override KustoConfiguration DockerResourceConfiguration { get; }
+
+    /// <inheritdoc />
+    public override KustoContainer Build()
+    {
+        Validate();
+        return new KustoContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override KustoBuilder Init()
+    {
+        return base.Init()
+            .WithImage(KustoImage)
+            .WithPortBinding(KustoPort, true)
+            .WithEnvironment("ACCEPT_EULA", "Y")
+            .WithWaitStrategy(Wait.ForUnixContainer().AddCustomWaitStrategy(new WaitUntil()));
```

Comment:
Can't you use the  HTTP wait strategy? E.g. like

    _ = Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request => request.ForPort(KustoPort).ForPath("/v1/rest/mgmt"));
    
(Probably won't work if the HTTP request requires a payload)

Reply:
Cool 😎, thanks for pointing that out.

You are right about the payload, so I extended the `HttpWaitStrategy`: https://github.com/testcontainers/testcontainers-dotnet/pull/963/commits/23f3c7526af05582e4f7a2f51ce94de77d93c9f3
With this extension I can use the same request (which I took from the offical documentation): https://github.com/testcontainers/testcontainers-dotnet/pull/963/commits/ad9635f67a4ecd4da2751593b148f86510efdec0

What do you think about this?

Reply:
Exactly what I thought. Unfortunately, it looks like we cannot do it like that. The HTTP client disposes the HTTP content if the first request fails. Afterward, we cannot use the content again 😕.

Reply:
I've added a suggestion that uses the container's console output to  detect readiness. Please share your thoughts on this. If you find it unreliable or prefer the curl approach, we can also go with that.

Reply:
> Exactly what I thought. Unfortunately, it looks like we cannot do it like that. The HTTP client disposes the HTTP content if the first request fails. Afterward, we cannot use the content again 😕.

Oh you are right (for .NET Core <= 3.0) according to this: https://makolyte.com/csharp-disposing-the-request-httpcontent-when-using-httpclient/
Not the first time I fall into this trap 😅

I changed the HttpWaitStrategy to take a factory for the `HttpContent`, so that we can handle the disposal. What do you think about this?: https://github.com/testcontainers/testcontainers-dotnet/pull/963/commits/09b0b0692b61470f62b5f93bf415ea9642f484a4

Reply:
> I've added a suggestion that uses the container's console output to detect readiness. Please share your thoughts on this. If you find it unreliable or prefer the curl approach, we can also go with that.

Thanks for the proposal 👍. I changed it back to the `HttpWaitStrategy` which now should be able to handle the diposal of the content correctly.

I thought it would be more stable to use the HTTP call which they also use in the documentation here: https://learn.microsoft.com/en-us/azure/data-explorer/kusto-emulator-install#install-the-kusto-emulator

Probably it is unlikely that they will ever change the console output and your proposal is much simpler, so I have no strong opinion on that. If you like checking the console output better, I can also revert the changes in the `HttpWaitStrategy` and go for your proposal.

Reply:
I think that's a good idea. I will take a closer look later the day.